### PR TITLE
CRM-19556 : Allow to search on just being an active member

### PR DESCRIPTION
**Pull request by liedekef**
- [CRM-19556: Allow to search on active membership](https://issues.civicrm.org/jira/browse/CRM-19556)

### CODE REVIEW

File: CRM/Member/BAO/Query.php [hunk 1]
```diff
@@ -251,6 +258,13 @@ public static function whereClauseSingle(&$values, &$query) {
         $query->_tables['civicrm_membership'] = $query->_whereTables['civicrm_membership'] = 1;
         return;
 
+      case 'membership_is_current_member':
+        // We don't want to include all tests for sql OR CRM-7827
+        $query->_where[$grouping][] = CRM_Contact_BAO_Query::buildClause("civicrm_membership_status.is_current_member", $op, $value, "Boolean");
+        $query->_qill[$grouping][] = ts('Active Member');
```
Comment (seamuslee001) [inline CRM/Member/BAO/Query.php]: only thing this should change if $value = 0 then it should be Inactive Member

Comment (monishdeb) [inline CRM/Member/BAO/Query.php]: something like 
``` 
$query->_qill[$grouping][] = ts('Is a Active Member? %1', array(1  => $value ? 'Yes' : 'No'));
```

Comment (liedekef) [inline CRM/Member/BAO/Query.php]: Check my latest commit and see if that is ok

**Comment by monishdeb** [comment]
@liedekef there is a style warning 
```
Query.php:264, SpaceBeforeDoubleArrow, Priority: High
Expected 1 space between "1" and double arrow; 2 found
```
highlighted in the related test build failure https://test.civicrm.org/job/CiviCRM-Core-PR/12408/

**Comment by monishdeb** [comment]
Changes look good, tested on local 

- For Active membership 
image

- For Inactive membership
image

**Comment by liedekef** [comment]
Ok, I'll fix the styling issues asap. Does the "merge on pass" mean that the PR will be immediately merged if all checks have passed?

**Comment by monishdeb** [comment]
Yup exactly  :) As when you fix the style error and push commit, test will re-build and if everything goes well it will be merged by me or other civicrm mergers.

**Comment by agh1** [comment]
I opened #9457 to address some wording issues.

**Pull request merged**